### ISSUE TITLE
PR reviewer: narration eval skill + harness

### ISSUE THREAD

**Issue by fairchild** [fairchild/workspaces#546]
Convert the recent PR reviewer narration analysis into a small skill-backed preference eval harness for fine-tuning narrative, evidence, and label behavior.

## Background

The managed PR reviewer now has three judgment surfaces beyond code review: `## Project Thread` narration, evidence sufficiency, label reasoning. Tuning is currently conversational + ad hoc. Need a repeatable way to collect reviewer output, present pairwise preference items, convert feedback into narrow prompt/context refinements.

The skill is the **lab** (gather examples, produce preference items, recommend prompt changes). The deployed reviewer still receives a trusted static system prompt + server-fetched context from `web/src/lib/agent-runtime/pr-review.ts` — no runtime skill loading.

## Key decisions

| Decision | Choice | Rationale |
|---|---|---|
| Runtime boundary | Keep deployed prompt in `pr-review.ts`; no skill loading at runtime | Managed Agents don't auto-load SKILL.md; loading from PR branch = prompt injection risk |
| Skill role | Repo skill for evaluation/improvement, not live review | Workflow is repeatable, judgment-heavy, benefits from stable rubric |
| Eval format | Small pairwise (A/B/tie/neither) per dimension | Easier to align than reviewing entire PR review bodies |
| Dimensions | `thread_narrative`, `evidence_judgement`, `label_reasoning` | Recently-added reviewer behaviors |
| Initial scope | Seed from PRs #431, #433–#443 | Positive examples + evidence-boundary cases + label suggestions + new-thread cases |
| Change policy | Eval feedback → prompt/context fine-tuning only until repeated failures justify code change | Avoid overfitting a reviewer that's already mostly working |

## Phases

1. **Skill skeleton.** `.agents/skills/pr-reviewer-narration-evals/` — SKILL.md, references/rubric.md, templates/preference-item.json. Skill clearly states it's *not* loaded by the production reviewer.
2. **Collector script.** Single-file UV script: `--repo`, `--prs 431,433-443` or `--limit N`; fetches PR metadata + managed reviews via `gh`; extracts `## Project Thread`, `## Evidence`, label rationale; writes deterministic JSONL to `evals/pr-reviewer-narration/raw.jsonl`.
3. **Human review harness.** `items.jsonl` (one dimension per item), `review-sheet.md` (plain Markdown), `preferences.jsonl` (append-only). Harness can later swap into `skill-creator`'s `eval-viewer/generate_review.py`.
4. **Convert preferences → prompt recommendations.** Likely candidates:
   - Relationship taxonomy: `causal`, `same surface`, `coordination overlap`, `new root thread`
   - Treat "new thread" as successful outcome when no prior PR clearly related
   - Supplement updated-order candidate scanning with same-label/same-path candidates
   - Re-check comments when prior review requested evidence + later comment claims evidence added
   - Tighten language around when structural proof is enough without hosted evidence

## Item shape

```json
{
  "id": "pr-443-thread-v1",
  "dimension": "thread_narrative",
  "current_pr": {...},
  "key_context": {"recent_prs": [...], "policy": "..."},
  "outcome_a": {"source": "actual-review", "text": "..."},
  "outcome_b": {"source": "candidate-rewrite", "text": "..."},
  "preference": null, "rationale": null, "tags": []
}
```

Tags start small: `wrong_prior`, `right_prior_weak_explanation`, `new_thread_handled_well`, `too_strict`, `too_lenient`, `evidence_good`, `evidence_bad`, `label_over_applied`, `label_under_applied`, `label_good`.

## References

- `web/src/lib/agent-runtime/pr-review.ts` — production prompt
- `web/src/lib/agent-runtime/__tests__/pr-review.test.ts` — contract tests
- `.agents/skills/skill-creator/SKILL.md` — skill creation + human eval flow
- Pairs with the continuous-reruns sibling plan

Source: backlog/pr-reviewer-narration-eval-skill_plan.md.
